### DISCUSSION TITLE
Missing variable is: CMAKE_Fortran_PREPROCESS_SOURCE

### DISCUSSION THREAD

**Discussion by gouff** [scivision/mumps-superbuild#78] [Q&A]
PS C:\repo\mumps-5.6.2.2> cmake -G Ninja -B build
-- The C compiler identification is Clang 18.1.0
-- The Fortran compiler identification is unknown
Intel(R) Fortran Compiler for applications running on Intel(R) 64, Version 2024.0.2 Build 20231213
Copyright (C) 1985-2023 Intel Corporation. All rights reserved.

-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/llvm-mingw-20240306-ucrt-x86_64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working Fortran compiler: C:/Program Files (x86)/Intel/oneAPI/2024.0/bin/ifx.exe
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_Fortran_PREPROCESS_SOURCE
CMake Error at C:/cmake-3.25.3-windows-x86_64/share/cmake-3.29/Modules/CMakeTestFortranCompiler.cmake:49 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  CMakeLists.txt:16 (project)

-- Configuring incomplete, errors occurred!

**Comment by scivision** [comment]
you need to use the Intel oneAPI Command Prompt from the Start => Programs => Intel oneAPI menu, otherwise the Intel oneAPI compiler isn't detected properly as you're experiencing. In general MUMPS works with oneAPI including on Windows.
